### PR TITLE
WC: Products Page

### PR DESCRIPTION
**Pull request by Prabhav-Sharma**
### Products Page
I have converted my products screen to react, created separate components for product cards. Added products provider to be able to use products anywhere in the app

Other Changes
- Created helper functions to filter data on the products page, namely Category, price, rating, sortBy.
- Added EMI tenure changing functionality.
- Added Products data in the database
- Shifted Component styling for CarouselCard and Arrival Card in their respective folders
- Added productsReducer to better manage state across the app

Files to check

- [ ] src/components/Card/ProductCard/ProductCard.jsx
- [ ] src/pages/Products/Products.jsx
- [ ] src/contexts/Providers/ProductProvider/ProductProvider.js
- [ ] src/contexts/Providers/ProductProvider/helpers.js
- [ ] src/contexts/Providers/ProductProvider/productReducer.js
- [ ] src/index.js   (Added ProductProvider)
- [ ] src/backend/db/products.js  (One need only check a single data for schema)

PS- Ignore other files

All CSS Files have been reviewed earlier in this repositiory [Car-Smart](https://github.com/Prabhav-Sharma/Car-Smart)

Preview:
chrome-capture (4)

### CODE REVIEW

**Review by Mohitkumar33 — APPROVED** [review]
Nice

**Review by AmishMishra11 — APPROVED** [review]
- Very descriptive pr
- good folder structure
- clean code

File: src/pages/Products/Products.jsx [hunk 1]
```diff
@@ -1,9 +1,293 @@
...
+    category,
+    sortBy,
+    rating,
+    EMI,
+    filterPrice,
+  } = state;
+
+  useEffect(() => {
+    document.title = "Products | CarSmart";
+  }, []);
```
Comment (devananth) [inline src/pages/Products/Products.jsx]: You can *create a custom hook* for this as you might want to change the *document.title* for every page. ( kindof: **useDocumentTitle** )

**Review by devananth — CHANGES_REQUESTED** [review]
- Very **descriptive PR**
   - Added a **suggestion** , have a look at it.
   - Good Work !

File: src/pages/Products/Products.jsx (hunk 1 above)
Comment (Prabhav-Sharma) [reply src/pages/Products/Products.jsx]: Yes, I'd be in need of that, I've added the custom hook. Please review

**Review by devananth — APPROVED** [review]
Good to **merge** !

**Pull request merged**